aospy/utils/times.py

```diff
@@ -497,3 +497,40 @@ def assert_matching_time_coord(arr1, arr2):
                'arr1: {}\narr2: {}')
     if not (arr1[TIME_STR].identical(arr2[TIME_STR])):
         raise ValueError(message.format(arr1[TIME_STR], arr2[TIME_STR]))
+
+
+def ensure_time_as_dim(ds):
+    """Ensures that time is an indexable dimension on relevant quantites
+
+    In xarray, scalar coordinates cannot be indexed.  We rely
+    on indexing in the time dimension throughout the code; therefore
+    we need this helper method to (if needed) convert a scalar time coordinate
+    to a dimension.
+
+    Parameters
+    ----------
+    ds : Dataset
+        Dataset with a time coordinate
+
+    Returns
+    -------
+    Dataset
+    """
+    TIME_STR = internal_names.TIME_STR
+    GRID_ATTRS_NO_TIMES = internal_names.GRID_ATTRS.copy()
+    del GRID_ATTRS_NO_TIMES[TIME_STR]
```

Comment:
I was going to say that the `pop` method is more idiomatic here, but this SO question seems to imply otherwise: http://stackoverflow.com/questions/27155819/delete-a-key-and-value-from-an-ordereddict

Not a big deal either way